app/components/GeneralErrorBoundary.tsx

```
@@ -56,8 +58,20 @@ export function GeneralErrorBoundary({
   statusHandlers?: Record<number, StatusHandler>
   unexpectedErrorHandler?: (error: unknown) => JSX.Element | null
 }): JSX.Element {
-  const error = useRouteError()
-  const params = useParams()
+  // Try to read router hooks; fall back gracefully when not in a Router
+  let error: unknown = null
+  let params: Record<string, string | undefined> = {}
+  try {
+    // The test suite often mocks this hook; calling it is safe even without Router
+    error = useRouteError()
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)